File: src/main/java/me/zeroeightsix/kami/module/modules/misc/AutoReply.java
package me.zeroeightsix.kami.module.modules.misc;

import java.util.function.Predicate;
import me.zeroeightsix.kami.util.Wrapper;
import net.minecraft.network.play.server.SPacketChat;
import me.zero.alpine.listener.EventHandler;
import me.zeroeightsix.kami.event.events.PacketEvent;
import me.zero.alpine.listener.Listener;
import me.zeroeightsix.kami.module.Module;

@Module.Info(name = "AutoReply", description = "automatically replies to private messages", category = Module.Category.MISC)
public class AutoReply extends Module
{
    @EventHandler
    public Listener<PacketEvent.Receive> receiveListener;
    
    public AutoReply() {
        this.receiveListener = new Listener<PacketEvent.Receive>(event -> {
            if (event.getPacket() instanceof SPacketChat && ((SPacketChat)event.getPacket()).getChatComponent().getUnformattedText().contains("whispers:")) {
                Wrapper.getPlayer().sendChatMessage("/r Kami Blue On Top");
            }
        }, (Predicate<PacketEvent.Receive>[])new Predicate[0]);
    }
}

File: common/src/main/kotlin/com/lambda/module/modules/player/AutoReply.kt
package com.lambda.module.modules.player

import com.lambda.event.events.PacketEvent
import com.lambda.event.listener.SafeListener.Companion.listener
import com.lambda.module.Module
import com.lambda.module.tag.ModuleTag
import com.lambda.util.Formatting.string
import com.lambda.util.StringUtils.capitalize
import net.minecraft.network.packet.s2c.play.GameMessageS2CPacket
import net.minecraft.util.math.Vec3d

object AutoReply : Module(
    name = "AutoReply",
    description = "Automatically Replies to messages",
    defaultTags = setOf(ModuleTag.PLAYER)
){
    private val range by setting("Reply range", 5000, 0..25000, 250, "Replies with coords if player is within range to spawn", " Blocks")
    init{
        listener<PacketEvent.Receive.Pre>{ event ->
            val packet = event.packet
            if (packet !is GameMessageS2CPacket) return@listener
            val msg = packet.content.string
            if (!msg.contains("Coords", true)) return@listener
            if (!msg.contains(" whispers: ")) return@listener
            if (!player.pos.isInRange(Vec3d.ZERO, range.toDouble())) return@listener
            val sender = msg.split(" whispers: ").first()
            connection.sendChatCommand("w $sender ${player.blockPos.string} [${world.dimensionKey.value.path.capitalize()}]")
        }
    }
}